WooCommerce/Classes/ServiceLocator/ProductImageUploader.swift

```diff
@@ -1,9 +1,21 @@
+import Combine
 import struct Yosemite.ProductImage
 import enum Yosemite.ProductAction
 import protocol Yosemite.StoresManager
 
+/// Information about an update of product image upload.
+struct ProductImageUploadUpdate {
+    let siteID: Int64
+    let productID: Int64
+    let productImageStatuses: [ProductImageStatus]
+    let error: Error?
+}
+
 /// Handles product image upload to support background image upload.
 protocol ProductImageUploaderProtocol {
+    /// Emits product image upload status updates (e.g. when the status changes or an error occurs).
```

Comment:
great question (and other similar questions below)! it was indeed quite confusing before, since we were still considering showing a notice for successful product updates before. Now that we decided not to (ref p91TBi-8Tm-p2#comment-10723), I renamed everything "status updates" to "errors" in https://github.com/woocommerce/woocommerce-ios/pull/7145/commits/30024d07e328e15f5ed981985aa0ea81031982f8